Topics: Add Swap-in Potentiam (SIP) (#1538)
bitcoinops/bitcoinops.github.io: _topics/en/swap-in-potentiam.md

  trust requirements by precommitting to an LSP and incorporating a timeout
  mechanism for unilateral access.

## Optional.  Produces a Markdown link with either "[title][]" or
## "[title](link)"
primary_sources:
  - title: Moving Onchain Funds "Instantly" To Lightning (Jesse Posner, ZmnSCPxj)
    link: https://lists.linuxfoundation.org/pipermail/lightning-dev/2023-January/003810.html

  - title: LSPS C= Extension, Swap-in-Potentiam
    link: https://github.com/ZmnSCPxj-jr/swap-in-potentiam/blob/master/doc/swap-in-potentiam.md

## Optional.  Each entry requires "title", "url", and "date".  May also use "feature:
## true" to bold entry
optech_mentions:
  - title: "Swap-in Potentiam Announced"
    url: /en/newsletters/2023/12/20/#swap-in-potentiam

  - title: Non-interactive LN channel open commitments
    url: /en/newsletters/2023/01/11/#non-interactive-ln-channel-open-commitments

## Optional.  Same format as "primary_sources" above
see_also:
  - title: Hash Time Locked Contracts from Bitcoin Wiki
    link: https://en.bitcoin.it/wiki/Hash_Time_Locked_Contracts

  - title: Swap-in potentiam for Phoenix
    link: https://gist.github.com/t-bast/5fd89979a8088b99d0b95c124902aa56

---

> SIP is a [submarine swap](https://bitcoinops.org/en/topics/submarine-swaps/)

Reviewer: ```suggestion
SIP is a [submarine swap][topic submarine swaps]
```

mechanism that differs from typical purely HTLC based submarine swaps by
allowing for a pre-commital stage. This allows for reusable swap addresses and
more flexibility with onchain funds.

In this scheme, the reusable swap address is a contract between Alice and Bob
that allows for two spend paths; Alice and Bob cosign, or timeout before Alice
can spend unilaterally.

Once the swap address is funded and confirmed, Alice can choose one of the
following options; swap the funds into a new or existing lightning network
channel with the help of Bob, or send the funds to a new onchain address with
the help of Bob. Alice also has the third option of waiting for the recovery
timeout if Bob is offline or uncooperative.

SIP is primarily geared towards mobile users who may go offline for extended
periods of time. The main trick of SIP is splitting the confirmation
requirements between a pre-commital address and a commited address, allowing

> for the immediate usage of funds when the mobile user comes online.

Reviewer: ```suggestion
for the immediate usage of funds when the mobile user comes online.

{% include references.md %}

```

---
title: Swap-in Potentiam (SIP)

## Optional. test Shorter name to use for reference style links e.g., "foo"
## will allow using the link [topic foo][].  Not case sensitive
shortname: swap-in potentiam

## Optional.  An entry will be added to the topics index for each alias
#aliases:
#  - Foo

## Required.  At least one category to which this topic belongs.  See
## schema for options
categories:
  - Contract Protocols
  - Lightning Network

## Required.  Use Markdown formatting.  Only one paragraph.  No links allowed.
## Should be less than 500 characters
excerpt: >
  **Swap-in-Potentiam (SIP)** is a protocol designed to enable the immediate transfer
  of already-confirmed onchain Bitcoin funds to the Lightning Network, lowering

>   trust requirements by precommitting to an LSP and incorporating a timeout

Reviewer 2: "Lowering trust requirements" compared to what?  (I'm guessing compared to zero-conf channels that don't use SIP.)  Perhaps an alternative phrasing could be: "...to the Lightning Network, allowing channels to be opened without either delay or counterparty risk by precommitting..."

Reviewer 2: Right, but I think we need to explain to the reader what baseline trust model we're comparing this to is when we make a claim that this lowers trust requirements.  SIP isn't lower trust than typical channel opens (which wait for multiple confirmations) or onchain spending (which also waits for multiple confirmations); it's only lower trust than an LSP who would otherwise accept zero-conf or low-conf channels.

  mechanism for unilateral access.

## Optional.  Produces a Markdown link with either "[title][]" or
## "[title](link)"
primary_sources:
  - title: Moving Onchain Funds "Instantly" To Lightning (Jesse Posner, ZmnSCPxj)
    link: https://lists.linuxfoundation.org/pipermail/lightning-dev/2023-January/003810.html

  - title: LSPS C= Extension, Swap-in-Potentiam
    link: https://github.com/ZmnSCPxj-jr/swap-in-potentiam/blob/master/doc/swap-in-potentiam.md

## Optional.  Each entry requires "title", "url", and "date".  May also use "feature:
## true" to bold entry
optech_mentions:
  - title: "Swap-in Potentiam Announced"
    url: /en/newsletters/2023/12/20/#swap-in-potentiam

  - title: Non-interactive LN channel open commitments

>     url: /en/newsletters/2023/01/11/#non-interactive-ln-channel-open-commitments

Reviewer 2: We have a soft policy [1] that a subject should have at least three mentions before we create a topic page for it.  I think you can add this as a third link, if you want: https://bitcoinops.org/en/newsletters/2023/07/19/#phoenix-wallet-adds-splicing-support

[1] "because this is an index to the Optech website (and not a guide to the Bitcoin system in general), we currently have a policy that a topic should be mentioned in at least three separate Optech newsletters or blog posts before we create a topic page for it."  https://bitcoinops.org/en/topics-announcement/

Reviewer 2: We don't currently add transcript mentions to topic entries.  I think we'll add that in the future in a separate section after the transition to Hugo.

## Optional.  Same format as "primary_sources" above
see_also:
  - title: Hash Time Locked Contracts from Bitcoin Wiki

>     link: https://en.bitcoin.it/wiki/Hash_Time_Locked_Contracts

Reviewer 2: Is there a special reason to link to HTLCs on the wiki here rather than our [HTLCs topic](https://bitcoinops.org/en/topics/htlc/) (which itself links to the wiki)?

  - title: Swap-in potentiam for Phoenix
    link: https://gist.github.com/t-bast/5fd89979a8088b99d0b95c124902aa56

---

> SIP is a [submarine swap][topic submarine swaps]

Reviewer 2: Is it?  As I think about it, a submarine swap starts with Alice and Bob both already having channels; Alice creates a UTXO that Bob can only spend if by revealing a preimage that will allow Alice to claim sats in her channel.

A SIP can start with neither party having a channel and only one party contributing any funding.  The second party doesn't give the first party anything (except a signature, if the first party wants to fully open the channel).  That doesn't seem like a "swap" to me at all.  I think it's just an alternative channel opening protocol.

Reviewer 2: Mermaidjs is not supported.  You can draw graphs in any tool, export or screenshot them, and include a PNG.  Several articles do this, you can `cd _topics/en/ ; git grep '\.png'` for examples.

I still don't think SIPs are a type of submarine swap, but if that's a claim supported by its authors, I'm fine with us saying that.

Reviewer 3: I am also surprised by the categorization of a swap-in potentiam as a Submarine Swap. Maybe the definition of Submarine Swaps has evolved, but my understanding of Submarine Swaps was that they allowed the first or the last hop of a multi-hop payment to be executed on-chain, and it involved the on-chain funds to be locked in an HTLC construction.

Reviewer 3: I still don’t see how it would be beneficial to group these together. 

FWIU, SIP refers to receiving an on-chain payment in a way that stages funds for a later splice-in or channel opening operation because they are already temporarily co-owned by the channel co-owners (or to-be channel co-owners). I would be open to an argument that splicing is similar to submarine swaps, but SIP seems to be a separate, removed step from it.

> mechanism that differs from typical purely HTLC based submarine swaps by

Reviewer 4: ```suggestion
mechanism that differs from typical purely HTLC-based submarine swaps by
```

allowing for a pre-commital stage. This allows for reusable swap addresses and
more flexibility with onchain funds.

In this scheme, the reusable swap address is a contract between Alice and Bob
that allows for two spend paths; Alice and Bob cosign, or timeout before Alice
can spend unilaterally.

Once the swap address is funded and confirmed, Alice can choose one of the
following options; swap the funds into a new or existing lightning network
channel with the help of Bob, or send the funds to a new onchain address with
the help of Bob. Alice also has the third option of waiting for the recovery
timeout if Bob is offline or uncooperative.

SIP is primarily geared towards mobile users who may go offline for extended
periods of time. The main trick of SIP is splitting the confirmation

> requirements between a pre-commital address and a commited address, allowing

Reviewer 4: ```suggestion
requirements between a pre-commital address and a committed address, allowing
```

> for the immediate usage of funds when the mobile user comes online.

Reviewer 2: I don't understand what you mean by "splitting the confirmation requirements between a pre-committal address and a committed address".

Reviewer 2: If this sentence just restates the previous paragraph, we can probably drop it and eliminate any confusion it causes.

Reviewer 3: I also find this sentence more confusing than helpful.

---
title: Swap-in Potentiam (SIP)

## Optional. test Shorter name to use for reference style links e.g., "foo"
## will allow using the link [topic foo][].  Not case sensitive
shortname: swap-in potentiam

## Optional.  An entry will be added to the topics index for each alias
#aliases:
#  - Foo

## Required.  At least one category to which this topic belongs.  See
## schema for options
categories:
  - Contract Protocols
  - Lightning Network

## Required.  Use Markdown formatting.  Only one paragraph.  No links allowed.
## Should be less than 500 characters
excerpt: >
  **Swap-in-Potentiam (SIP)** is a protocol that facilitates the immediate transfer of confirmed
  on-chain Bitcoin to the Lightning Network. It reduces trust requirements compared to other

>   instant channel opening methods like 0-conf channels by pre-committing to a Lightning Service
>   Provider (LSP) and incorporating a timeout mechanism for unilateral access.

Reviewer 3: How about:

```suggestion
  instant channel opening methods like 0-conf channels by temporarily committing to co-ownership with a Lightning Service
  Provider (LSP) and delaying unilateral access per a timeout.
```

## Optional.  Produces a Markdown link with either "[title][]" or
## "[title](link)"
primary_sources:
  - title: Moving Onchain Funds "Instantly" To Lightning (Jesse Posner, ZmnSCPxj)
    link: https://lists.linuxfoundation.org/pipermail/lightning-dev/2023-January/003810.html

>   - title: LSPS C= Extension, Swap-in-Potentiam

Reviewer: @arshbot in going through the content one last time before merge, I saw this is a dead link and it doesn't appear there is an equivalent repo under https://github.com/ZmnSCPxj-jr/

Is there another different link you are aware of? Or should we remove this reference?

Reviewer: Dropped this link for now.

    link: https://github.com/ZmnSCPxj-jr/swap-in-potentiam/blob/master/doc/swap-in-potentiam.md

## Optional.  Each entry requires "title", "url", and "date".  May also use "feature:
## true" to bold entry
optech_mentions:
  - title: "Swap-in Potentiam Announced"
    url: /en/newsletters/2023/12/20/#swap-in-potentiam

  - title: Non-interactive LN channel open commitments
    url: /en/newsletters/2023/01/11/#non-interactive-ln-channel-open-commitments

## Optional.  Same format as "primary_sources" above
see_also:
  - title: Hash Time Locked Contracts from Bitcoin Wiki
    link: https://en.bitcoin.it/wiki/Hash_Time_Locked_Contracts

  - title: Swap-in potentiam for Phoenix
    link: https://gist.github.com/t-bast/5fd89979a8088b99d0b95c124902aa56

  - title: Payjoin-in-Potentiam
    link: https://delvingbitcoin.org/t/payjoin-in-potentiam-externally-fund-an-lsp-channel-open-with-one-transaction/749

---
SIP is a [submarine swap][topic submarine swaps]
mechanism that differs from typical purely HTLC-based submarine swaps by

> allowing for a pre-commital stage. This allows for reusable swap addresses and

Reviewer 3: Why is it important for the address to be reusable? Isn’t that a privacy hazard?

> more flexibility with onchain funds.

Reviewer 3: How does locking up your funds with a prospective channel partner provide more flexibility? It provides an option on a channel open or splice, but wouldn’t the shared control make the funds _less flexible_ for all other purposes?

> In this scheme, the reusable swap address is a contract between Alice and Bob

Reviewer 3: If the "reusable" aspect of the address is important, it should perhaps be further elaborated.

> that allows for two spend paths; Alice and Bob cosign, or timeout before Alice
> can spend unilaterally.

Reviewer 3: How about:

```suggestion
that allows for two spend paths; Alice and Bob cosign, or Alice
can spend unilaterally after a timeout.
```

---
title: Swap-in Potentiam (SIP)

## Optional. test Shorter name to use for reference style links e.g., "foo"
## will allow using the link [topic foo][].  Not case sensitive
shortname: swap-in potentiam

## Optional.  An entry will be added to the topics index for each alias
#aliases:
#  - Foo

## Required.  At least one category to which this topic belongs.  See
## schema for options

> categories:

Reviewer: ```suggestion
topic-categories:
```

  Provider (LSP) and delaying unilateral access per a timeout.

## Optional.  Produces a Markdown link with either "[title][]" or
## "[title](link)"
primary_sources:
  - title: Moving Onchain Funds "Instantly" To Lightning (Jesse Posner, ZmnSCPxj)
    link: https://lists.linuxfoundation.org/pipermail/lightning-dev/2023-January/003810.html

  - title: LSPS C= Extension, Swap-in-Potentiam
    link: https://github.com/ZmnSCPxj-jr/swap-in-potentiam/blob/master/doc/swap-in-potentiam.md

## Optional.  Each entry requires "title", "url", and "date".  May also use "feature:
## true" to bold entry
optech_mentions:
  - title: "Swap-in Potentiam Announced"
    url: /en/newsletters/2023/12/20/#swap-in-potentiam

  - title: Non-interactive LN channel open commitments
    url: /en/newsletters/2023/01/11/#non-interactive-ln-channel-open-commitments

## Optional.  Same format as "primary_sources" above
see_also:
  - title: Hash Time Locked Contracts from Bitcoin Wiki
    link: https://en.bitcoin.it/wiki/Hash_Time_Locked_Contracts

  - title: Swap-in potentiam for Phoenix
    link: https://gist.github.com/t-bast/5fd89979a8088b99d0b95c124902aa56

>   

Reviewer: ```suggestion

```

  - title: Swaproot
    link: https://acinq.co/blog/phoenix-swaproot

  - title: Payjoin-in-Potentiam
    link: https://delvingbitcoin.org/t/payjoin-in-potentiam-externally-fund-an-lsp-channel-open-with-one-transaction/749

---
Swap-in potentiam (SIP) differs from typical HTLC-based swaps by
allowing for a pre-commital stage. This allows for; reusable swap addresses,
immediate transfers to lightning, and reduced trust requirements compared to
0-conf channels.

In this scheme, the reusable swap address is a contract between Alice and Bob
that allows for two spend paths; Alice and Bob cosign, or Alice
can spend unilaterally after a timeout.

Once the swap address is funded and confirmed, Alice can choose one of the
following options; swap the funds into a new or existing lightning network
channel with the help of Bob, or send the funds to a new onchain address with
the help of Bob. Alice also has the third option of waiting for the recovery
timeout if Bob is offline or uncooperative.

SIP is primarily geared towards mobile users who may go offline for extended

> periods, and a variation called swaproot has been implemented in Pheonix wallet.

Reviewer 3: ```suggestion
periods, and a variation called swaproot has been implemented in Phoenix wallet.
```
